### PR TITLE
Include _id in migration helper

### PR DESCRIPTION
**Pull request by mrpasquini**
- [x] @aergonaut 
- [x] @johnny-lai 

Chris mentioned upcoming upgrade problems. We can help reduce the load by adding `_id` in the migration helper now.

### CODE REVIEW

**Comment by johnny-lai** [comment]
Does this only affect new tables?

**Comment by johnny-lai** [comment]
I'm wondering if this is safe... it looks like we use `userstamps` in migrations.

So if an existing customer on an old version upgrades with this code, he will run the existing migrations and the table will have `created_by_id`.

If a new customer is bootstrapped, he will run db/schema.rb, and would get `created_by`.

**Comment by aergonaut** [comment]
Maybe make it take a parameter to choose the ID style and just make sure that new migrations choose the new `_id` style. That way we wouldn't be retroactively changing the behavior of old migrations.

**Comment by mrpasquini** [comment]
Review Request Bump. I changed this to take a hash so we can pass a new arguement for including the _id column in compatibility mode.

File: lib/userstamp/migration_helper.rb [hunk 1]
```diff
@@ -6,10 +6,14 @@ def self.included(base) # :nodoc:
...
-          column(Ddb::Userstamp.compatibility_mode ? :updated_by : :updater_id, :integer)
-          column(Ddb::Userstamp.compatibility_mode ? :deleted_by : :deleter_id, :integer) if include_deleted_by
+        def userstamps(options = {})
+          # add backwards compatability support
+          options = {include_deleted_by: false} if options == false
+          id_string = options[:rails_5_compatibility_mode] ? '_id' : ''
+
+          column(Ddb::Userstamp.compatibility_mode ? "created_by#{id_string}".to_sym : :creator_id, :integer)
+          column(Ddb::Userstamp.compatibility_mode ? "updated_by#{id_string}".to_sym : :updater_id, :integer)
+          column(Ddb::Userstamp.compatibility_mode ? "deleted_by#{id_string}".to_sym : :deleter_id, :integer) if include_deleted_by
```
Comment (johnny-lai) [inline lib/userstamp/migration_helper.rb]: `include_deleted_by` is not set

File: lib/userstamp/migration_helper.rb [hunk 2]
```diff
@@ -6,10 +6,14 @@ def self.included(base) # :nodoc:
...
 
       module InstanceMethods
-        def userstamps(include_deleted_by = false)
-          column(Ddb::Userstamp.compatibility_mode ? :created_by : :creator_id, :integer)
-          column(Ddb::Userstamp.compatibility_mode ? :updated_by : :updater_id, :integer)
-          column(Ddb::Userstamp.compatibility_mode ? :deleted_by : :deleter_id, :integer) if include_deleted_by
+        def userstamps(options = {})
+          # add backwards compatability support
+          options = {include_deleted_by: false} if options == false
+          id_string = options[:rails_5_compatibility_mode] ? '_id' : ''
```
Comment (johnny-lai) [inline lib/userstamp/migration_helper.rb]: Let's just call this something like `append_id_suffix`? Adding `_id` is fine for Rails 4 as well.

**Comment by johnny-lai** [comment]
It looks like in the original version, it used to take a `true`/`false` argument.

`userstamps(true)` will generate 3 columns (created, updated and deleted)
`userstamps` or `userstamps(false)` will generate 2 columns (created, updated)

It feels like you are trying to accept the old API, in addition to a new, but as it is right now, I don't think it's accepting the old API correctly.

File: lib/userstamp/migration_helper.rb [hunk 3]
```diff
@@ -6,10 +6,14 @@ def self.included(base) # :nodoc:
       end
 
       module InstanceMethods
-        def userstamps(include_deleted_by = false)
-          column(Ddb::Userstamp.compatibility_mode ? :created_by : :creator_id, :integer)
-          column(Ddb::Userstamp.compatibility_mode ? :updated_by : :updater_id, :integer)
-          column(Ddb::Userstamp.compatibility_mode ? :deleted_by : :deleter_id, :integer) if include_deleted_by
+        def userstamps(options = {})
+          # add backwards compatability support
```
Comment (johnny-lai) [inline lib/userstamp/migration_helper.rb]: typo `compatability`

**Comment by johnny-lai** [comment]
lgtm

**Pull request merged**